internal/server/providers/oidc.go

```diff
@@ -38,6 +38,21 @@ type OIDCClient interface {
 	GetUserInfo(ctx context.Context, providerUser *models.ProviderUser) (*UserInfoClaims, error)
 }
 
+type key struct{}
+
+var ctxKey = key{}
```

Comment:
This is the recommended way of creating a key for storing things in `context.Context`. By creating a value of an unexported struct, no other place can read or set this value. It forces everything to use the two functions below, which return or accept the typed value.